scipy/stats/_continuous_distns.py

```diff
@@ -10481,41 +10504,51 @@ def _shape_info(self):
         return [ib, ic]
 
     def _argcheck(self, b, c):
-        b = float(b)  # ensure float
         return (b != 0.) & (c > 1.)
 
     def _get_support(self, b, c):
         return self.a, c
 
     def _pdf(self, x, b, c):
+        b = np.asarray(b, dtype=x.dtype)  # avoid int to negative int power
         return b * x**-(b+1) / (1 - 1/c**b)
 
     def _logpdf(self, x, b, c):
         return np.log(b) - np.log(-np.expm1(-b*np.log(c))) - (b+1)*np.log(x)
 
     def _cdf(self, x, b, c):
+        b = np.asarray(b, dtype=x.dtype)  # avoid int to negative int power
         return (1 - x**-b) / (1 - 1/c**b)
 
     def _logcdf(self, x, b, c):
+        b = np.asarray(b, dtype=x.dtype)  # avoid int to negative int power
         return np.log1p(-x**-b) - np.log1p(-1/c**b)
 
     def _ppf(self, q, b, c):
+        b = np.asarray(b, dtype=q.dtype)  # avoid int to negative int power
         return pow(1 - (1 - 1/c**b)*q, -1/b)
 
     def _sf(self, x, b, c):
+        b = np.asarray(b, dtype=x.dtype)  # avoid int to negative int power
         return (x**-b - 1/c**b) / (1 - 1/c**b)
 
     def _logsf(self, x, b, c):
+        b = np.asarray(b, dtype=x.dtype)  # avoid int to negative int power
         return np.log(x**-b - 1/c**b) - np.log1p(-1/c**b)
 
     def _isf(self, q, b, c):
+        b = np.asarray(b, dtype=q.dtype)  # avoid int to negative int power
         return pow(1/c**b + (1 - 1/c**b)*q, -1/b)
 
     def _entropy(self, b, c):
         return -(np.log(b/(1 - 1/c**b))
```

Comment:
Apparently, `entropy`'s arguments are converted to Python floats/ints. I have no idea why.